src/commands/deploy/deploy.ts

```diff
@@ -102,31 +106,26 @@ const getDeployFolder = async ({ command, config, options, site, siteData }) =>
         filter: (input) => resolve(command.workingDir, input),
       },
     ])
-    deployFolder = promptPath
+    deployFolder = promptPath as string
   }
 
   return deployFolder
 }
 
-/**
- * @param {string} deployFolder
- */
-// @ts-expect-error TS(7006) FIXME: Parameter 'deployFolder' implicitly has an 'any' t... Remove this comment to see the full error message
-const validateDeployFolder = async (deployFolder) => {
-  /** @type {import('fs').Stats} */
-  let stats
+const validateDeployFolder = async (deployFolder: string) => {
+  let stats: Stats
   try {
     stats = await stat(deployFolder)
   } catch (error_) {
-    // @ts-expect-error TS(2571) FIXME: Object is of type 'unknown'.
-    if (error_.code === 'ENOENT') {
-      return error(`No such directory ${deployFolder}! Did you forget to run a build?`)
-    }
+    if (error_ && typeof error_ === 'object' && 'code' in error_) {
+      if (error_.code === 'ENOENT') {
+        return error(`No such directory ${deployFolder}! Did you forget to run a build?`)
```

Comment:
This is not related to your PR, but I think this message could be improved.

```suggestion
        return error(`The deploy directory "${deployFolder}" has not been found. Did you forget to run 'netlify build'?`)
```

Reply:
I fear this will break now all the snapshots 😂